src/expr/src/id.rs

```diff
@@ -21,13 +21,21 @@ pub enum Id {
     Local(LocalId),
     /// An identifier that refers to a global dataflow.
     Global(GlobalId),
+    /// Bare sources don't (yet?) live in the same namespace as catalog items (including the final, possibly transformed source.)
+    /// This gives us a way to refer to them in relation expressions
+    BareSource(GlobalId),
+    /// Used to refer to a bare source within the RelationExpr defining the transformation of that source (before an ID has been
+    /// allocated for the bare source).
+    LocalBareSource,
```

Comment:
Hrm, this seems like a hack. Do you think we'll be able to get rid of this eventually? E.g. if in the catalog a source becomes both a bare source (with a global ID) and a view atop that source (with a global ID)?

Reply:
Yes, getting rid of it long-term is the plan.